src/amazonqFeatureDev/session/sessionState.ts

```diff
@@ -34,15 +34,24 @@ export class PrepareRefinementState implements Omit<SessionState, 'uploadId'> {
         this.tokenSource = new vscode.CancellationTokenSource()
     }
     async interact(action: SessionStateAction): Promise<SessionStateInteraction> {
-        const { zipFileBuffer, zipFileChecksum } = await prepareRepoData(this.config.sourceRoot, action.telemetry)
+        const uploadId = await telemetry.amazonq_createUpload.run(async span => {
```

Comment:
not sure if uploadId var is needed let' use try/catch here.
example: https://github.com/aws/aws-toolkit-vscode/blob/853b3f8de8d2907f323ea80948324ee6ce4da734/src/amazonqFeatureDev/session/sessionState.ts#L69

Reply:
I need the uploadId as an input in the RefinementState.

Reply:
I still would prefer to have try/catch inside iteract

Reply:
I canot wrap the whole interaction, otherwise it means also wrapping the next step constructor which is not needed in this case. It will also then mix metrics with the approachInoke as the RefinementState does that. better to leave them separated so we don't have duplicate metrics.